tiatoolbox/models/dataset/interactive_segmentation.py

```diff
@@ -0,0 +1,156 @@
+            resolution = self.resolution, units = self.units)
+
+    def __getitem__(self, idx):
+        patch = self.patch_extractor.__getitem__(idx)
+
+        boundingBox = self.get_boundingBox(idx)
+
+        # we know nucPoint is at the centre of the patch:
+        nucPoint = np.zeros((1, self.patch_size[1], self.patch_size[1]), dtype=np.uint8)
+        nucPoint[0,int((self.patch_size[1]-1)/2),int((self.patch_size[1]-1)/2)] = 1
+
+        exclusionMap = self.get_exclusionMap(idx, boundingBox)
+
+        patch = np.moveaxis(patch, 2, 0)
+        patch = patch / 255
+
+        input = np.concatenate((patch, nucPoint, exclusionMap), axis=0, dtype=np.float32)   # shape=(c=5,h,w)
+ 
+        data = {
+            "input": input,
+            "boundingBox": boundingBox,
+            "click": (self.locations["x"][idx], self.locations["y"][idx])
+        }
+        if self.label is not None:
+            data["label"] = self.label
+        
+        return data
+
+
+    def get_boundingBox(self, idx):
+        """This function returns a bounding box of size (patch_size x patch_size) that has the click as its centre.
+            The bounding box is the same box that is used in patch extraction.
+
+        Args:
+            idx (int): The index of the point ("Click") to get a bounding box for.
+        Returns:
+            bounds: a list of coordinates in `[start_x, start_y, end_x, end_y]`
+            format to be used for patch extraction.
+
+        """
+
+        #Coordinates of the top left corner of each patch:
+        location = (self.patch_extractor.locations_df["x"][idx], self.patch_extractor.locations_df["y"][idx])
+
+        tl = np.array(location)
+        br = location + np.array(self.patch_size)
+        bounds = np.concatenate([tl, br])
+
+        return  bounds
+    
+    def get_exclusionMap(self, idx, boundingBox):
+        """This function returns an exclusion map for click at the given index.
+
+        Args:
+            idx (int): The index of the point ("Click") to get an exclusionMap for.
+            boundingBox: a list of coordinates in `[start_x, start_y, end_x, end_y]`
+                This is the bounding box for the click at the given index.
+        Returns:
+            exclusionMap (ndarray)
+
+        """
+
+        # Exclude the current click from list of clicks:
+        otherPoints = self.locations.drop(idx, axis = 0)
+        x_locations = otherPoints["x"].to_numpy()
+        y_locations = otherPoints["y"].to_numpy()
+      
+        exclusionMap = np.zeros((1, self.patch_size[1], self.patch_size[1]), dtype=np.uint8)
+        
+        xStart = boundingBox[0]
+        yStart = boundingBox[1]
+        xEnd = boundingBox[2]
+        yEnd = boundingBox[3]
+
+        for i in range(x_locations.shape[0]):
+            x = x_locations[i]
+            y = y_locations[i]
+
+            if (x >= xStart and x <= xEnd and y >= yStart and y <= yEnd):
+                exclusionMap[0, x - xStart, y - yStart] = 1
```

Comment:
if you want to go with the for loop, make sure that x, y order in array indexing is right
For example, is it  exclusionMap[0, x, y] or exclusionMap[0, y, x]? to me, the second one should be the right one. You can test it and see if it generates reasonable output.

Reply:
Yeah you are right, it should be [0, y, x]